cpp/include/DataStorm/InternalT.h

```diff
@@ -41,31 +41,31 @@ namespace DataStormI
 
     template<typename T, typename Enabler = void> struct EncoderT
     {
-        static std::vector<unsigned char> encode(const std::shared_ptr<Ice::Communicator>&, const T& value)
+        static Ice::ByteSeq encode(const Ice::CommunicatorPtr&, const T& value)
```

Comment:
For later: we should check if we need a communicator when encoding. I suspect we don't.

Reply:
We support both, there are two versions of the Encoder, one accepts the communicator.